src/checkers/inference/util/SlotDefaultTypeResolver.java

```diff
@@ -190,7 +191,11 @@ public Void visitAnnotatedType(AnnotatedTypeTree tree, Void unused) {
         @Override
         public Void visitNewClass(NewClassTree tree, Void unused) {
             AnnotatedTypeMirror defaultType = getDefaultTypeFor(tree);
-            defaultTypes.put(tree.getIdentifier(), defaultType);
+            ExpressionTree type = tree.getIdentifier();
+            if ((type instanceof ParameterizedTypeTree) && !((ParameterizedTypeTree) type).getTypeArguments().isEmpty())
```

Comment:
I think you need to loop through the type arguments and put the corresponding types into `defaultTypes`. You can take `visitParameterizedType` as a reference.

Reply:
You're right, I forgot the `typeArguments` are not arguments to the class' type parameters. Those are arguments to type parameters on the constructor.